### PR TITLE
Fix partial rendering with dot in filename

### PR DESCRIPTION
**Pull request by bquorning**
When rendering a collection with a partial whose filename contains a dot, e.g. "customer.mobile", we would set a `locals[:'customer.mobile']` variable instead of, as in earlier versions of Rails, `locals[:customer]`.

This bug was introduced in da9038eaa5d19c77c734a044c6b35d7bfac01104.

### CODE REVIEW

**Comment by schneems** [comment]
Do you know why or when this behavior changed?

**Comment by bquorning** [comment]
Hey @schneems. This bug was introduced in da9038eaa5d19c77c734a044c6b35d7bfac01104.

**Comment by schneems** [comment]
@amatsuda can you take a look at this PR?

**Comment by amatsuda** [comment]
@bquorning Interesting. I've never seen such a code, and it's actually an untested (=undefined) behavior so far, but there's no reason not to keep the compatibility on your code.
Thank you for the fix! :heart:

**Comment by bquorning** [comment]
Thank you for the merge.

File: actionview/lib/action_view/renderer/partial_renderer.rb [hunk 1]
```diff
@@ -521,7 +521,7 @@ def retrieve_template_keys
     def retrieve_variable(path, as)
       variable = as || begin
         base = path[-1] == "/".freeze ? "".freeze : File.basename(path)
-        raise_invalid_identifier(path) unless base =~ /\A_?(.*)(?:\.\w+)*\z/
+        raise_invalid_identifier(path) unless base =~ /\A_?(.*?)(\.\w+)*\z/
```
Comment (bquorning) [inline actionview/lib/action_view/renderer/partial_renderer.rb]: Ah, sorry. I just now noticed that there’s really no need to remove `?:` from the last regex capture.

Comment (amatsuda) [inline actionview/lib/action_view/renderer/partial_renderer.rb]: True. I added the `?:` to express that we're only using `$1` afterwards.
It would work without `?:`, or you can make another PR adding `?:`, or I would just push these 2 chars.
I'm fine with any option :)

Comment (bquorning) [inline actionview/lib/action_view/renderer/partial_renderer.rb]: You can commit directly to master – that’s probably the easiest option :-)

Comment (amatsuda) [inline actionview/lib/action_view/renderer/partial_renderer.rb]: @bquorning Done. 28a6571bae1d6bc1c774e8e501720374061a31f1
Thanks!
